static/js/components/material/Dialog.js

```diff
@@ -0,0 +1,80 @@
+// @flow
+import React from 'react';
+import {MDCDialog} from '@material/dialog/dist/mdc.dialog';
+
+import Button from './Button';
+
+type DialogProps = {
+  open: boolean,
+  onAccept: () => void,
+  onCancel: () => void,
+  children: React.Children,
+  title: string,
+};
+
+export default class Dialog extends React.Component {
+  dialog: null;
+  dialogRoot: null;
+  props: DialogProps;
+
+  componentDidMount() {
+    const { open, onAccept, onCancel } = this.props;
+    this.dialog = new MDCDialog(this.dialogRoot);
+
+    this.dialog.listen('MDCDialog:accept', onAccept);
+    // $FlowFixMe: Flow thinks this.dialog might be null
+    this.dialog.listen('MDCDialog:cancel', onCancel);
+
+    if (open) {
+      // $FlowFixMe: Flow thinks this.dialog might be null
+      this.dialog.show();
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.dialog) {
+      this.dialog.destroy();
+    }
+  }
+
+  componentWillReceiveProps(nextProps: DialogProps) {
+    if (this.dialog) {
+      if (!this.props.open && nextProps.open) {
```

Comment:
Suggestion, does this maybe read a little easier? I had to read this over about twice to verify what it was doing and that it was doing it right.

```
if (this.props.open !== nextProps.open) {
  if (this.nextProps.open) {
    this.dialog.show();
  } else {
    this.dialog.close();
  }
}